strawberryfields/engine.py

```diff
@@ -336,8 +360,144 @@ def run(self, program, *, shots=1, compile_options={}, modes=None, state_options
             # empty sequence
             pass
         else:
-            result.state = self.backend.state(modes, **state_options)  # tfbackend.state can use kwargs
+            result.state = self.backend.state(
+                modes, **state_options
+            )  # tfbackend.state can use kwargs
         return result
 
 
+class StarshipEngine(BaseEngine):
+    """
+    Starship quantum program executor engine.
+
+    Executes :class:`.Program` instances on the chosen remote backend, and makes
+    the results available via :class:`.Result`.
+
+    Args:
+        backend (str, BaseBackend): name of the backend, or a pre-constructed backend instance
```

Comment:
Could we add support for also initializing the engine with API client arguments via keyword arguments, cf. ADR STRAW-92:

```python
eng = sf.HTTPEngine(backend="chip0", host="api.strawberryfields.ai", api_key="XXXXX")
```

Reply:
Added parameters that match APIClient parameters. Let me know what you think.

Reply:
:+1: